### PR TITLE
Mount delete, refresh and default to new folder for mount create

### PR DESCRIPTION
**Pull request by kevcunnane**
- Delete mount action added
- Refresh mount action added
- Added "mymount" to the end of existing path so that we don't use already-existing HDFS folder. The call fails unless folder doesn't exist

**Screenshots**
image
Refresh
image
Delete
image
Status:
image

### CODE REVIEW

File: extensions/big-data-cluster/src/bigDataCluster/dialog/mountHdfsDialog.ts [hunk 1]
```diff
@@ -125,16 +127,23 @@ abstract class HdfsDialogModelBase<T extends DialogProperties> {
...
-	constructor(props: MountHdfsProperties) {
-		super(props);
+	protected async createAndVerifyControllerConnection(): Promise<ClusterController> {
+		// We pre-fetch the endpoints here to verify that the information entered is correct (the user is able to connect)
+		let controller = this.createController();
+		let response: IEndPointsResponse;
+		try {
+			response = await controller.getEndPoints();
+			if (!response || !response.endPoints) {
+				throw new Error(localize('mount.hdfs.loginerror1', "Login to controller failed"));
```
Comment (alanrenmsft) [inline extensions/big-data-cluster/src/bigDataCluster/dialog/mountHdfsDialog.ts]: login error? copy-paste?

File: extensions/big-data-cluster/src/bigDataCluster/dialog/mountHdfsDialog.ts [hunk 2]
```diff
@@ -125,16 +127,23 @@ abstract class HdfsDialogModelBase<T extends DialogProperties> {
...
+			}
+		} catch (err) {
+			// An error here means we failed to
```
Comment (alanrenmsft) [inline extensions/big-data-cluster/src/bigDataCluster/dialog/mountHdfsDialog.ts]: seems the comment is not completed yet.

File: extensions/big-data-cluster/src/bigDataCluster/dialog/mountHdfsDialog.ts [hunk 3]
```diff
@@ -125,16 +127,23 @@ abstract class HdfsDialogModelBase<T extends DialogProperties> {
...
+			throw new Error(localize('mount.hdfs.loginerror2', "Login to controller failed: {0}", err.message));
```
Comment (alanrenmsft) [inline extensions/big-data-cluster/src/bigDataCluster/dialog/mountHdfsDialog.ts]: copy paste ?

File: extensions/big-data-cluster/package.json [hunk 4]
```diff
@@ -80,6 +88,16 @@
 					"command": "bigDataClusters.command.mount",
 					"when": "nodeType=~/^mssqlCluster/ && nodeType!=mssqlCluster:message && nodeSubType=~/^(?!:mount).*$/",
 					"group": "1mssqlCluster@10"
+				},
+				{
+					"command": "bigDataClusters.command.refreshmount",
+					"when": "nodeType == mssqlCluster:folder && nodeSubType=~/^:mount:$/",
```
Comment (Charles-Gagnon) [inline extensions/big-data-cluster/package.json]: Any reason you're using a regex here but not for delete?

Comment (kevcunnane) [reply extensions/big-data-cluster/package.json]: My mistake - I was doing A/B testing to see which one worked (both do). Will fix

File: extensions/big-data-cluster/src/bigDataCluster/dialog/mountHdfsDialog.ts (hunk 1 above)
Comment (kevcunnane) [reply extensions/big-data-cluster/src/bigDataCluster/dialog/mountHdfsDialog.ts]: I have 2 cases, 1 where I know the error message and one without. Hence the 2 messages loginerror1 and 2

**Pull request merged**